clouddrift/kinematics.py

```diff
@@ -3,9 +3,245 @@
+    latitude : array-like
+        Latitude sequence. Unidimensional array input.
+    relative_bandwidth : float
+        Bandwidth of the frequency-domain equivalent filter for the extraction of the inertial
+        oscillations; a number less or equal to one which is a fraction of the inertial frequency.
+        A value of 0.1 leads to a bandpass filter equivalent of +/- 10 percent of the inertial frequency.
+    time_step : float
+        The constant time interval between data points in seconds. Default is 3600.
+    relative_vorticity: Optional, float or array-like
+        Relative vorticity adding to the local Coriolis frequency. If "f" is the Coriolis
+        frequency then "f" + `relative_vorticity` will be the effective Coriolis frequency as defined by Kunze (1985).
+        Positive values correspond to cyclonic vorticity, irrespectively of the latitudes of the data
+        points.
+
+    Returns
+    -------
+    xhat : array-like
+        Zonal relative displacement in meters from inertial oscillations.
+    yhat : array-like
+        Meridional relative displacement in meters from inertial oscillations.
+
+    Examples
+    --------
+    To extract displacements from inertial oscillations from sequences of longitude
+    and latitude values, equivalent to bandpass around 20 percent of the local inertial frequency:
+
+    >>> xhat, yhat = extract_inertial_from_position(longitude, latitude, 0.2)
+
+    Next, the residual positions from the inertial displacements can be obtained with another function:
+
+    >>> residual_longitudes, residual_latitudes = residual_positions_from_displacements(longitude, latitude, xhat, yhat)
+
+    Raises
+    ------
+    ValueError
+        If longitude and latitude arrays do not have the same shape.
+        If relative_vorticity is an array and does not have the same shape as longitude and latitude.
+        If relative_bandwidth is not a float.
+        If time_step is not a float.
+        If relative_bandwidth is not less or equal to one.
+
+    See Also
+    --------
+    :func:`residual_positions_from_displacements`, `wavelet_transform`, `morse_wavelet`
+
+    """
+    if longitude.shape != latitude.shape:
+        raise ValueError("longitude and latitude arrays must have the same shape.")
+
+    # length of data sequence
+    data_length = longitude.shape[0]
+
+    if isinstance(relative_vorticity, float):
+        relative_vorticity = np.full_like(longitude, relative_vorticity)
+    elif isinstance(relative_vorticity, np.ndarray):
+        if not relative_vorticity.shape == longitude.shape:
+            raise ValueError(
+                "relative_vorticity must be a float or the same shape as longitude and latitude."
+            )
+
+    if not isinstance(relative_bandwidth, float):
+        raise ValueError("relative_bandwidth must be a float.")
+
+    if not relative_bandwidth <= 1:
+        raise ValueError("relative_bandwidth must be less or equal to one.")
```

Comment:
@selipot why check for `float` here if it needs to be <= 1?

Below we use `np.abs(relative_bandwidth)`, so are negative values OK?

If not, I suggest just checking for the value:

```suggestion
    if not 0 <= relative_bandwidth <= 1:
        raise ValueError("relative_bandwidth must be in the [0, 1] range.")
```

Reply:
Blame GitHub copilot :)

Reply:
0 will not work for a number of reasons.